src/utils/event-decorators.js

```diff
@@ -0,0 +1,163 @@
+/*
+ * Provides a decorator functor for annotating functions in Component definitions so that they are automatically
+ * bound and unbound from events without needing to call el.addEventListener or el.removeEventListener yourself.
+ *
+ * Decorations should be made in the Components definition object. Here is the simplest example:
+ *
+ * var {bindEvent} = require('event-binder');
+ *
+ * AFRAME.registerComponent("foo", {
+ *   componentchanged: bindEvent( function(evt) {
+ *     console.log(evt.detail);
+ *   })
+ * })
+ *
+ * By default the function will be bound to events corresponding to its property name, in this case: 'componentchanged'.
+ * It will listen for events on its parent element, and will begin listening or end listening when init or remove is
+ * called. However this can also be configured by passing a configuration object:
+ *
+ * AFRAME.registerComponent("foo", {
+ *   whenSceneLoads: bindEvent( {
```

Comment:
I don't think we need this customizable of an API. Should just take a function, listen to `this.el`, default to the property name, and set up on init/remove.

Reply:
I agree, this API seems too high of an abstraction for the core. I would probably keep it as a 3rd party component / library

Reply:
Not what I meant. I mean the extra options are not needed. The utility still belongs in core, optional to use. I use it extremely often not to have to bind methods, add event listeners up top, and have to manually remove event listeners on `pause` or `remove`, which no one ever does because it's too much work.